bin/faucet/README.md

```diff
@@ -46,5 +46,138 @@ The faucet implements several security measures to prevent abuse:
+
+```rust
+let base_url = "http://localhost:8080";
+let url = format!("{base_url}/pow?account_id=mlcl1qq8mcy8pdvl0cgqfkjzf8efjjsnlzf7q");
+let response = reqwest::get(&url).await?.error_for_status()?.text().await?;
+let json: serde_json::Value = serde_json::from_str(&response)?;
+let challenge: &str = json["challenge"].as_str().unwrap();
+let target: u64 = json["target"].as_u64().unwrap();
+```
+
+2. **Solve the computational challenge**
+
+```rust
+use sha3::{Digest, Sha3_256};
+
+let nonce = {
+    let mut found_nonce = None;
+    for nonce in 0..u64::MAX {
+        // Create SHA3-256 hash
+        let mut hasher = Sha3_256::new();
+        hasher.update(challenge.as_bytes());
+        hasher.update(nonce.to_be_bytes());
+        let hash = hasher.finalize();
+
+        // Take the first 8 bytes and interpret as big-endian u64
+        let number = u64::from_be_bytes(hash[..8].try_into().unwrap());
+
+        // Check if hash number is less than target
+        if number < target {
+            found_nonce = Some(nonce);
+            break;
+        }
+    }
+    found_nonce.expect("No valid nonce found")
+};
+```
+
+3. **Request tokens** along with your solved challenge to `/get_tokens`
+
+```rust
+let base_url = "http://localhost:8080";
+let params = format!("account_id=mlcl1qq8mcy8pdvl0cgqfkjzf8efjjsnlzf7q\
+    &is_private_note=true\
+    &asset_amount=100\
+    &challenge={}\
+    &nonce={}", challenge, &nonce.to_string());
+let url = format!("{base_url}/get_tokens?{params}");
+let response = reqwest::get(&url).await?.error_for_status()?;
+let text = response.text().await?;
+let json: serde_json::Value = serde_json::from_str(&text)?;
+```
+
+4. **Request note** to download generated private notes
+
+```rust
+use base64::Engine;
+use base64::engine::general_purpose;
+
+let base_url = "http://localhost:8080";
+let url = format!("{base_url}/get_note?note_id={note_id}");
+let response = reqwest::get(&url).await?.error_for_status()?;
+let text = response.text().await?;
+let json: serde_json::Value = serde_json::from_str(&text)?;
+
+// Decode note with base64 
+let note_data = general_purpose::STANDARD
+    .decode(json["data_base64"].as_str().unwrap())
+    .unwrap();
+std::fs::write("note.mno", &note_data).unwrap();
+```
+
+#### Example
+
+To see a full working example, check [request_tokens.rs](bin/faucet/examples/request_tokens.rs). The example assumes you have the faucet running on `http://localhost:8080`.
+
+Run it with:
+```bash
+cargo run --example request_tokens
+```
+
+#### API Endpoints Reference
+
+**GET /pow**
+- **Purpose**: Request a proof-of-work challenge
+- **Query Parameters**:
+  - `account_id` (string, required): The account ID requesting the challenge
+  - `api_key` (string, optional): API key for authentication
+- **Response**: JSON object containing:
+  - `challenge` (string): The encoded challenge string in hexadecimal format. This string serializes the target, timestamp, account ID, API key, and signature
```

Comment:
I'd probably delete the second sentence from here. I don't think we need to explain what goes into the challenge.